pkg/provider/remote/remote.go

```diff
@@ -0,0 +1,352 @@
+		principals := []*deploy.Resource{}
+		resources := []*deploy.Resource{}
+
+		for _, r := range v.Resources {
+			resources = append(resources, &deploy.Resource{
+				Name: r.Name,
+				Type: r.Type,
+			})
+		}
+
+		for _, p := range v.Principals {
+			principals = append(principals, &deploy.Resource{
+				Name: p.Name,
+				Type: p.Type,
+			})
+		}
+
+		req.Spec.Resources = append(req.Spec.Resources, &deploy.Resource{
+			Type: v1.ResourceType_Policy,
+			Config: &deploy.Resource_Policy{
+				Policy: &deploy.Policy{
+					Principals: principals,
+					Actions:    v.Actions,
+					Resources:  resources,
+				},
+			},
+		})
+	}
+
+	for k, v := range proj.ApiDocs {
+		apiBody, err := json.Marshal(v)
+		if err != nil {
+			return nil
+		}
+
+		req.Spec.Resources = append(req.Spec.Resources, &deploy.Resource{
+			Name: k,
+			Type: v1.ResourceType_Api,
+			Config: &deploy.Resource_Api{
+				Api: &deploy.Api{
+					Document: &deploy.Api_Openapi{
+						Openapi: string(apiBody),
+					},
+				},
+			},
+		})
+	}
+
+	req.Attributes = map[string]string{
+		"x-nitric-project": proj.Name,
+	}
+
+	for k, v := range stackCfg {
+		switch k {
+		case "name":
+			req.Attributes["x-nitric-stack"] = v.(string)
+		default:
+			req.Attributes[k] = fmt.Sprintf("%v", v)
+		}
+	}
+
+	return req
+}
+
+func projToDownRequest(proj *project.Project, stackCfg map[string]any, providerName string) *deploy.DeployDownRequest {
+	req := &deploy.DeployDownRequest{
+		Attributes: map[string]string{},
+	}
+
+	req.Attributes = map[string]string{
+		"x-nitric-project": proj.Name,
+	}
+
+	for k, v := range stackCfg {
+		switch k {
+		case "name":
+			req.Attributes["x-nitric-stack"] = v.(string)
+		default:
+			req.Attributes[k] = fmt.Sprintf("%v", v)
+		}
+	}
+
+	return req
+}
+
+func (p *remoteDeployment) Up(log output.Progress) (*types.Deployment, error) {
+	sc, err := stackConfig(p.proj.Dir, p.stackName, p.provider)
+	if err != nil {
+		return nil, err
+	}
+
+	client := deploy.NewDeployServiceClient(p.conn)
+
+	op, err := client.Up(context.Background(), projToUpRequest(p.proj, sc, p.provider))
+	if err != nil {
+		return nil, err
+	}
+
+	res := &types.Deployment{
+		Summary:      &types.Summary{},
+		ApiEndpoints: map[string]string{},
+	}
+
+	for {
+		evt, err := op.Recv()
+		if err != nil {
+			if errors.Is(err, io.EOF) {
+				return res, nil
+			}
+
+			return res, err
+		}
+
+		switch t := evt.Content.(type) {
+		case *deploy.DeployUpEvent_Message:
+			log.Debugf(t.Message.Message)
+		case *deploy.DeployUpEvent_Result:
+			if !t.Result.Success {
+				return res, errors.New("failed to deploy")
+			}
+			/*	for k, v := range res.Outputs {
+					if strings.HasPrefix(k, "api:") {
+						d.ApiEndpoints[strings.TrimPrefix(k, "api:")] = fmt.Sprint(v.Value)
+					}
+				}
+			*/
+
+			return res, nil
+		}
+	}
+}
+
+func (p *remoteDeployment) List() (interface{}, error) {
+	return []stackSummary{}, errors.New("not supported for remote deployments")
```

Comment:
Lets remove stackSummary for now and return nil here. If we need to keep stack listing we can raise requirements for the deploy contract against the nitric repo.